modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v1/form_data.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  module PdfConstruction
+    module NoticeOfDisagreement::V1
+      class FormData
+        def initialize(notice_of_disagreement)
+          @notice_of_disagreement = notice_of_disagreement
+        end
+
+        def veteran_name
+          name('Veteran')
+        end
+
+        def veteran_ssn
+          header_field_as_string('X-VA-Veteran-SSN')
+        end
+
+        def veteran_file_number
+          header_field_as_string('X-VA-Veteran-File-Number')
+        end
+
+        def veteran_dob
+          dob 'Veteran'
+        end
+
+        def mailing_address_number_and_street
+          'USE ADDRESS ON FILE'
+        end
+
+        def homeless?
+          notice_of_disagreement.veteran_homeless_state ? 1 : 'Off'
+        end
+
+        def preferred_phone
+          'USE PHONE NUMBER ON FILE'
+        end
+
+        def preferred_email
+          'USE EMAIL ON FILE'
+        end
+
+        def direct_review?
+          board_review_option == 'direct_review' ? 1 : 'Off'
+        end
+
+        def evidence_submission?
+          board_review_option == 'evidence_submission' ? 1 : 'Off'
+        end
+
+        def hearing?
+          board_review_option == 'hearing' ? 1 : 'Off'
+        end
+
+        def extra_contestable_issues?
+          contestable_issues.size > 5 ? 1 : 'Off'
+        end
+
+        def soc_opt_in?
+          notice_of_disagreement.form_data&.dig('data', 'attributes', 'socOptIn') ? 1 : 'Off'
+        end
```

Comment:
Should these be predicate methods?

Reply:
Are you saying that since they're not actually returning a true Boolean value ('Off' instead of false), they shouldn't have the `?`? I figured the form field is basically a true false value in the form of a checkbox, but I'm not sure now.

Reply:
I'm having the same question. Can't decide whether the FormData or the Structure should take care of knowing exactly how to tie the model's data into the pdf fields :(

Reply:
I'm thinking the predicate style makes sense in the context of the form... _But_ I'm not sure if it makes sense overall, for example if someone in the future was writing some logic around the form data and they attempted to do something like:


```
if direct_review?
  # ...
end
```

I'm torn about this... I like it in this immediate context, but I think it could be somewhat confusing too...

Reply:
I think it's worth pointing out that the `FormData` class is specific to data _for the form_. The value methods contained in that class are specifically to answer questions about the form in question, which, in this case is `NoticeOfDisagreement::V1`